lib/engine/connection.rb

```diff
@@ -21,16 +19,21 @@ def self.connect!(hex)
           hex = path.hex
           connection = connection.branch!(path)
           next if connection.paths.include?(path)
-          # FIXME: for intra-node paths
           next if connection.nodes.include?(path.nodes[0])
+          next if connection.nodes.include?(path.nodes[1])
           next if connection.paths.any? { |p| p.hex == hex && (p.exits & path.exits).any? }
 
           connections[connection] = true
           connection.add_path(path)
 
-          path.exits.each do |edge|
-            hex_connections = hex.connections[edge]
+          if path.exits.none?
```

Comment:
empty?, none? would be if you expect nils inside

Reply:
OK. Still learning the array/hash methods.